openml/flows/sklearn_converter.py

```diff
@@ -501,11 +501,26 @@ def _extract_information_from_model(model):
     for k, v in sorted(model_parameters.items(), key=lambda t: t[0]):
         rval = sklearn_to_flow(v, model)
 
+        def flatten_all(list_):
+            """ Flattens arbitrary depth lists of lists. """
+            for el in list_:
+                if isinstance(el, (list, tuple)):
+                    yield from flatten_all(el)
+                else:
+                    yield el
+
+        if isinstance(rval, (list, tuple)):
+            nested_list_of_simple_types = all([isinstance(el, (bool, str, int, float))
```

Comment:
Could you please add a few comments here on what you are doing here? Even though I know the issue it tries to prevent I'm having trouble understanding this.

Reply:
I am not sure specifically if you mean line 513, or also the method above. Line 513 makes sure that each element within the list of lists is itself a 'simple type' that we can serialize (e.g. `bool`, `int`, ...). Rather than it needing special treatment according to the first case of the if/else statement directly below (which serializes e.g. steps in a pipeline). Though this did remind me that since #635 is merged, I should also allow for numpy types in 513. Update coming (will add comments too).